contracts/SCBridgeWallet.sol

```diff
@@ -125,7 +135,26 @@ contract SCBridgeWallet is IAccount {
     challengeExpiry = largestTimeLock + CHALLENGE_WAIT;
   }
 
-  function execute(address dest, uint256 value, bytes calldata func) external {
+  /// crossChain is a special function that handles cross chain execution and payment
+  function crossChain(
+    ExecuteChainInfo[] calldata e,
+    PaymentChainInfo[] calldata p
+  ) public {
+    // Only the entrypoint should trigger this by excecuting a UserOp
+    require(msg.sender == entrypoint, "account: not EntryPoint");
```

Comment:
Curious if this is necessary -- should we have had this check before? Should all 4337 wallets have this check?

Reply:
It also seems to be replicated below?

Reply:
This is needed since we want to enforce that `crossChain` is only triggered through submitting a UserOp. This enforces that the only way to trigger `crossChain` is through a fully signed UserOp.

If we didn't have this check `require` then anyone could call with `crossChain` with whatever `PaymentChainInfo` or `ExecutionChainInfo` they wanted. This could be used to raise a challenge on-chain with any unsigned state.

Reply:
>It also seems to be replicated below?

I think that's in the `execute` function, which has a slightly less strict check.